addon/components/snippet-plugin/snippet-insert.ts

```diff
@@ -36,15 +36,30 @@ export default class SnippetInsertComponent extends Component<Args> {
   @action
   onInsert(content: string) {
     const domParser = new DOMParser();
+    const parsed = domParser.parseFromString(content, 'text/html').body;
+    const documentDiv = parsed.querySelector('div[data-say-document="true"]');
 
     this.closeModal();
 
-    this.controller.withTransaction((tr) => {
-      return tr.replaceSelectionWith(
-        ProseParser.fromSchema(this.controller.schema).parse(
-          domParser.parseFromString(content, 'text/html'),
+    if (documentDiv) {
+      return this.controller.withTransaction((tr) =>
+        tr.replaceSelectionWith(
+          ProseParser.fromSchema(this.controller.schema).parse(documentDiv, {
+            preserveWhitespace: true,
+            topNode: this.controller.schema.nodes.doc.create({
+              lang: documentDiv.getAttribute('lang') ?? undefined,
+            }),
```

Comment:
This is not necessary as we're are not parsing the document node (we want to prevent it in this case)